libyul/backends/wasm/polyfill/Comparison.yul

```diff
@@ -48,11 +48,7 @@ function eq(x1, x2, x3, x4, y1, y2, y3, y4) -> r1, r2, r3, r4 {
 
 // returns 0 if a == b, -1 if a < b and 1 if a > b
 function cmp(a, b) -> r:i32 {
-	switch i64.lt_u(a, b)
-	case 1:i32 { r := 0xffffffff:i32 }
-	default {
-		r := i64.ne(a, b)
-	}
+	r := i32.select(0xffffffff:i32, i64.ne(a, b), i64.lt_u(a, b))
```

Comment:
Once we get to the point of having simplification rules for wasm, this could simplified if the conditions is a constant :)

Reply:
Also the code generator for `switch` and `if` can benefit from `select` maybe.

Reply:
I think it would be easier having an optimisation step for that (translate any 2-case switch which assigns to the same, single variable).